org.eclipse.jdt.ls.product/scripts/jdtls.py

```diff
@@ -20,17 +20,21 @@
 import tempfile
 
 def get_java_executable(known_args):
+	is_windows = platform.system() == 'Windows'
 	if known_args.java_executable is not None:
 		java_executable = known_args.java_executable
 	else:
 		java_executable = 'java'
 
 		if 'JAVA_HOME' in os.environ:
-			ext = '.exe' if platform.system()  == 'Windows' else ''
+			ext = '.exe' if is_windows else ''
 			java_exec_to_test = Path(os.environ['JAVA_HOME']) / 'bin' / f'java{ext}'
 			if java_exec_to_test.is_file():
 				java_executable = java_exec_to_test.resolve()
 
+	if is_windows:
+		java_executable = str(java_executable)
+
```

Comment:
Instead of this, would it still work if you just called `str(java_executable)` immediately after the resolve call from above. The `get_java_executable` seems to always return a string except in the above case.

Reply:
Ah, I wasn't sure the intended behavior is to always return a string (it's clear now that you mention it). I've moved the `str()` call to immediately surround `java_exec_to_test.resolve()` now.